File: BigFootVentures/BigFootVentures.Business.Objects/Management/Cancellation.cs
namespace BigFootVentures.Business.Objects.Management
{
    public sealed class Cancellation : BusinessBase
    {
        #region "Properties"

        public string OwnerName { get; set; }

        public string ReferenceInternal { get; set; }
        public string ReferenceExternal { get; set; }
        public string SentOrigin { get; set; }
        public string ReceiptDate { get; set; }
        public string PaymentDate { get; set; }
        public string FilingCost { get; set; }
        public Trademark Trademark { get; set; }
        public string Status { get; set; }
        public string InternalCaseNumber { get; set; }
        public string SubmissionMethod { get; set; }
        public Company Applicant { get; set; }
        public string PaymentReference { get; set; }
        public string ResearchPerformance { get; set; }
        public string Comments { get; set; }

        public string AdmissibleDate { get; set; }
        public string OwnerResponseDeadline { get; set; }
        public string ObservationDeadline { get; set; }
        public string OwnerResponseDeadlineExt { get; set; }
        public string ObservationDeadlineExt { get; set; }

        public string AcquisitionLetterSentOrigin { get; set; }
        public string AcquisitionLetterSentMethod { get; set; }
        public string AcquisitionLetterDateSent { get; set; }
        public string OwnerResponseAcquisitionLetter { get; set; }
        public bool DomainEnquiry { get; set; }
        public string DomainEnquiryNotes { get; set; }
        public string Outcome { get; set; }
        public string OutcomeDate { get; set; }
        public string WithdrawalReason { get; set; }
        public string TrademarkAssignmentDate { get; set; }
        public string TrademarkAcquisitionCost { get; set; }
        public string UDRPStrategy { get; set; }
        public string UDRPStrategyComment { get; set; }
        public string NegotiationComments { get; set; }
        public string WithdrawalDate { get; set; }
        public string ExpenseClaimReceivedDate { get; set; }
        public string CostRefund { get; set; }

        #endregion
    }
}

File: BigFootVentures/BigFootVentures.Application.Web/Scripts/Views/Shared/_Components/Modals/AdvanceSearch/_Cancellation.js
function BuildCancellationQuery(query) {
    var referenceInternal = $('[name="Input.Filter.Cancellation.ReferenceInternal"]').val();
    var referenceExternal = $('[name="Input.Filter.Cancellation.ReferenceExternal"]').val();
    var sentOrigin = $('[name="Input.Filter.Cancellation.SentOrigin"]').val();
    var internalCaseNumber = $('[name="Input.Filter.Cancellation.InternalCaseNumber"]').val();
    var submissionMethod = $('[name="Select.Filter.Cancellation.SubmissionMethod"]').val();
    var applicant = $('[name="Input.Filter.Cancellation.Applicant"]').val();
    var trademark = $('[name="Input.Filter.Cancellation.Trademark"]').val();
    var researchPerformance = $('[name="Select.Filter.Cancellation.ResearchPerformance"]').val();
    var status = $('[name="Select.Filter.Cancellation.Status"]').val();
    var acquisitionLetterSentOrigin = $('[name="Select.Filter.Cancellation.AcquisitionLetterSentOrigin"]').val();
    var acquisitionLetterSentMethod = $('[name="Select.Filter.Cancellation.AcquisitionLetterSentMethod"]').val();
    var UDRPStrategy = $('[name="Select.Filter.Cancellation.UDRPStrategy"]').val();
    var ownerResponseAcquisitionLetter = $('[name="Select.Filter.Cancellation.OwnerResponseAcquisitionLetter"]').val();
    var domainEnquiry = $('[name="Select.Filter.Cancellation.DomainEnquiry"]').val();
    var outcome = $('[name="Select.Filter.Cancellation.Outcome"]').val();

    if (referenceInternal != null && referenceInternal != undefined && referenceInternal.length > 0) {
        query += '&referenceInternal=' + referenceInternal;
    }

    if (referenceExternal != null && referenceExternal != undefined && referenceExternal.length > 0) {
        query += '&referenceExternal=' + referenceExternal;
    }

    if (sentOrigin != null && sentOrigin != undefined && sentOrigin.length > 0) {
        query += '&sentOrigin=' + sentOrigin;
    }

    if (internalCaseNumber != null && internalCaseNumber != undefined && internalCaseNumber.length > 0) {
        query += '&internalCaseNumber=' + internalCaseNumber;
    }

    if (submissionMethod != '') {
        query += '&submissionMethod=' + submissionMethod;
    }

    if (applicant != null && applicant != undefined && applicant.length > 0) {
        query += '&applicant=' + applicant;
    }

    if (trademark != null && trademark != undefined && trademark.length > 0) {
        query += '&trademark=' + trademark;
    }

    if (researchPerformance != '') {
        query += '&researchPerformance=' + researchPerformance;
    }

    if (status != '') {
        query += '&status=' + status;
    }

    if (acquisitionLetterSentOrigin != '') {
        query += '&acquisitionLetterSentOrigin=' + acquisitionLetterSentOrigin;
    }

    if (acquisitionLetterSentMethod != '') {
        query += '&acquisitionLetterSentMethod=' + acquisitionLetterSentMethod;
    }

    if (UDRPStrategy != '') {
        query += '&UDRPStrategy=' + UDRPStrategy;
    }

    if (ownerResponseAcquisitionLetter != '') {
        query += '&ownerResponseAcquisitionLetter=' + ownerResponseAcquisitionLetter;
    }

    if (domainEnquiry != 'NotSet') {
        query += '&domainEnquiry=' + domainEnquiry;
    }

    if (outcome != '') {
        query += '&outcome=' + outcome;
    }

    return query;
}